File: Symulator Świata/guarana.py
from plant import Plant
STRENGTH_INCREASE = 3


class Guarana(Plant):

    def __init__(self, world, coord=None):
        self.name = 'u'
        self._strength = 0
        self._initiative = 0
        self._age = 0
        self.sprite = (190, 30, 35)
        self.world = world
        if coord is not None:
            self.world.board[coord] = self
        else:
            self.spawn()

    def hasIncreasedStrength(self, coord):
        self.world.commentator.addComment("{0} zwieksza sile o {1}".format(self.world.board[coord].name,
                                                                           STRENGTH_INCREASE))
        self.world.board[coord].strength += STRENGTH_INCREASE
        return True

    def createChild(self, coord):
        g = Guarana(self.world, coord)

File: PO PROJEKT 1/PO PROJEKT 1/PROJEKT PO PLIKI ŹRÓDŁOWE/Guarana.cpp
#include "Guarana.h"

Guarana::Guarana(World* world) {
	increaseStrength(0);
	setInitiative(0);
	sprite = 'u';
	this->world = world;
	spawn();
}

Guarana::Guarana(World* world, Coordinate coord) {
	increaseStrength(0);
	setInitiative(0);
	sprite = 'u';
	this->world = world;
	world->board.insert({ coord, this });
}

Organism* Guarana::createChild(Coordinate coord) {
	return new Guarana(world, coord);
}

bool Guarana::hasIncreasedStrength(Coordinate coord) {
	world->board[coord]->increaseStrength(STRENGTH_INCREASE);
	cout << world->board[coord]->getSprite() << " zwieksza swoja sile o " << STRENGTH_INCREASE << endl;
	return true;
}

void Guarana::draw() {
	cout << GREEN_BCKG << RED << getSprite() << RESET;
}
